File: system-test/dotnet/SystemTests/Latest/AcceptanceTests/CancelOrderPositiveIsolatedTest.cs
using SystemTests.Latest.AcceptanceTests.Base;
using Dsl.Core.Shop;
using Driver.Port.Shop.Dtos;
using Optivem.Testing;
using Xunit;

namespace SystemTests.Latest.AcceptanceTests;

[Collection("Isolated")]
public class CancelOrderPositiveIsolatedTest : BaseAcceptanceTest
{
    [Theory]
    [Time]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    [ChannelInlineData("2024-12-31T21:59:59Z")]
    [ChannelInlineData("2024-12-31T22:30:01Z")]
    [ChannelInlineData("2024-12-31T10:00:00Z")]
    [ChannelInlineData("2025-01-01T22:15:00Z")]
    public async Task ShouldBeAbleToCancelOrderOutsideOfBlackoutPeriod31stDecBetween2200And2230(Channel channel, string timeIso)
    {
        await Scenario(channel)
            .Given().Clock().WithTime(timeIso)
            .And().Order().WithStatus(OrderStatus.Placed)
            .When().CancelOrder()
            .Then().ShouldSucceed();
    }
}

File: system-test/java/src/test/java/com/optivem/shop/systemtest/latest/acceptance/CancelOrderPositiveIsolatedTest.java
package com.optivem.shop.systemtest.latest.acceptance;

import com.optivem.shop.systemtest.latest.acceptance.base.BaseAcceptanceTest;
import com.optivem.shop.testkit.channel.ChannelType;
import com.optivem.shop.testkit.driver.port.shop.dtos.OrderStatus;
import com.optivem.testing.*;
import org.junit.jupiter.api.TestTemplate;

@Isolated
class CancelOrderPositiveIsolatedTest extends BaseAcceptanceTest {
    @TimeDependent
    @TestTemplate
    @Channel(value = {ChannelType.API}, alsoForFirstRow = ChannelType.UI)
    @DataSource({"2024-12-31T21:59:59Z"})   // 1 second before blackout period starts
    @DataSource({"2024-12-31T22:30:01Z"})   // 1 second after blackout period ends
    @DataSource({"2024-12-31T10:00:00Z"})   // Another time on blackout day but outside blackout period
    @DataSource({"2025-01-01T22:15:00Z"})   // Another day entirely (same time but different day)
    void shouldBeAbleToCancelOrderOutsideOfBlackoutPeriod31stDecBetween2200And2230(String timeIso) {
        scenario
                .given().clock()
                    .withTime(timeIso)
                .and().order()
                    .withStatus(OrderStatus.PLACED)
                .when().cancelOrder()
                .then().shouldSucceed();
    }
}
